src/Engine/GameInstance.java

```diff
@@ -283,7 +291,18 @@ public void endGame()
       army.deInitForGame(this);
     }
   }
-  
+
+  /**
+   * Gives you a random number between 0 (inclusive) and the input (exclusive)
+   * <p>Call this in GameAction.getEvents()/event constructors and nowhere else
+   */
+  public int getRN(int upperBound)
+  {
+    upperBound = Math.abs(upperBound);
+    ++rngNumbersGenerated;
+    return rng.nextInt(upperBound);
+  }
```

Comment:
I have a bit of heartburn creating a public, effectively global method that is only supposed to be used by one component. How do we either a) prevent misuse or b) detect it if it happens?

Reply:
We already had that discussion on the other PR, remember? I just forgot to rebase.

Reply:
I do remember the discussion - I don't remember which PR it was, or if I asked exactly these questions, or what the outcome was.

Reply:
That'll be this boi: https://github.com/ThislsAUsername/DefendPeace/pull/382#discussion_r1252276588